### PR TITLE
feat(auth): make a subset of auth apis runnable with the server context

### PR DESCRIPTION
**Pull request by HuiSF**
### `@amplify/core`
* Removed `**/package.json` from the `files` field in the `package.json`
* Now export the low level `fetchAuthSession` from the `/internals` subpath

### `@amplify/auth`
Made the following APIs to be callable with Amplify server context.
* `getCurrentUser`
* `fetchUserAttributes`
* Updated relevant unit tests

#### Description of how you validated changes

* Unit tests
* Making API call in Next.js server.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by stocaaro — APPROVED** [review]
How do we unit test this type of behavior for the server?

File: packages/auth/src/providers/cognito/apis/server/fetchUserAttributes.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+import {
+	AmplifyServer,
+	getAmplifyServerContext,
+} from '@aws-amplify/core/lib-esm/adapterCore';
+import { AuthUserAttribute } from '../../../../types';
+import { CognitoUserAttributeKey } from '../../types';
+import { fetchUserAttributes as fetchUserAttributesInternal } from '../internal/fetchUserAttributes';
+
+export const fetchUserAttributes = (
```
Comment (israx) [inline packages/auth/src/providers/cognito/apis/server/fetchUserAttributes.ts:12]: should we add doc strings?

Comment (HuiSF) [reply packages/auth/src/providers/cognito/apis/server/fetchUserAttributes.ts:12]: Yup, I will add the docs to all server side API as a whole in a separate PR.

**Comment by HuiSF** [comment]
> How do we unit test this type of behavior for the server?

The only difference between server and client is where the Amplify instance comes from. The existing unit tests has been verifying the common share internal logic, so an extra set of unit tests probably unnecessary.

I will add unit tests to all server side APIs to verify `getAmplifyServerContext` function at least in a separate PR.

**Review by israx — DISMISSED** [review]

**Review by stocaaro — DISMISSED** [review]

File: packages/auth/package.json [hunk 2]
```diff
@@ -88,7 +88,7 @@
 		"@aws-amplify/core": "^6.0.0"
 	},
 	"devDependencies": {
-		"@aws-amplify/core": "6.0.0",
+		"@aws-amplify/core": "^6.0.0",
```
Comment (jimblanc) [inline packages/auth/package.json:91]:
```suggestion
		"@aws-amplify/core": "6.0.0",
```

The dev dep shouldn't be pinned, it's for our internal dev and lerna will handle it

File: packages/core/internals/package.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+	"name": "@aws-amplify/core/internals",
```
Comment (jimblanc) [inline packages/core/internals/package.json:2]: Do we need a new path for this? Could we possible re-use `@aws-amplify/core/internals/utils`?

**Review by jimblanc — CHANGES_REQUESTED** [review]
Looks good, one blocking comment on version range

**Review by jimblanc — APPROVED** [review]

**Review by israx — APPROVED** [review]

**Pull request merged**